src/java.desktop/share/classes/javax/swing/DefaultListSelectionModel.java

```diff
@@ -686,23 +690,27 @@ public void insertIndexInterval(int index, int length, boolean before)
      * the selection model.  This is typically called to sync the selection
      * model width a corresponding change in the data model.  Note
      * that (as always) index0 need not be &lt;= index1.
+     *
+     * @throws IndexOutOfBoundsException if either index is negative
+     *         or equal to Integer.MAX_VALUE
      */
     public void removeIndexInterval(int index0, int index1)
     {
+        if (index0 < 0 || index1 < 0) {
+            throw new IndexOutOfBoundsException("index is negative");
+        }
+        if (index0 == Integer.MAX_VALUE || index1 == Integer.MAX_VALUE) {
+            throw new IndexOutOfBoundsException("index is equal to MAX_VALUE");
+        }
         int rmMinIndex = Math.min(index0, index1);
         int rmMaxIndex = Math.max(index0, index1);
-        int gapLength = ((rmMaxIndex - rmMinIndex) + 1) > (rmMaxIndex - rmMinIndex)
-                ? ((rmMaxIndex - rmMinIndex) + 1) : (rmMaxIndex - rmMinIndex);
+        int gapLength = (rmMaxIndex - rmMinIndex) + 1;
```

Comment:
```suggestion
        int gapLength = (rmMaxIndex - rmMinIndex) == Integer.MAX_VALUE
                        ? Integer.MAX_VALUE
                        : (rmMaxIndex - rmMinIndex) + 1;
```
This ensures `gapLength` remains positive. (The condition `(rmMaxIndex - rmMinIndex) == Integer.MAX_VALUE` is `true` if and only if `rmMinIndex == 0 && rmMaxIndex == Integer.MAX_VALUE`.)